### PR TITLE
Allow filtering the "Students" main reporting table by Course or Memb…

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
…ership

## Description
Add filtering to the "Students" main reporting table to allow filtering that screen by Course or Membership

## How has this been tested?
Played a little bit with the filter and made sure it worked as expected.

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests.
- [x] My code follows the LifterLMS Coding Standards.

### CODE REVIEW

**Comment by eri-trabiccolo** [comment]
@thomasplevy I've thought to treat this separately as it doesn't depend on https://github.com/gocodebox/lifterlms/pull/835

File: includes/admin/reporting/tables/llms.table.students.php [hunk 1]
```diff
@@ -284,6 +336,11 @@ private function get_query_args() {
 
 			$query_args['statuses'] = array( $this->get_filter() );
 
+		} elseif ( 'course_membership' === $this->get_filterby() && '' !== $this->get_filter() ) {
```
Comment (thomasplevy) [inline includes/admin/reporting/tables/llms.table.students.php]: update method changelog

File: includes/admin/reporting/tables/llms.table.students.php [hunk 2]
```diff
@@ -417,6 +474,11 @@ protected function parse_args( $args = array() ) {
 
 		$this->per_page = isset( $args['per_page'] ) ? $args['per_page'] : $this->get_per_page();
 
+		if ( $this->is_filterable ) {
```
Comment (thomasplevy) [inline includes/admin/reporting/tables/llms.table.students.php]: update method changelog

**Comment by eri-trabiccolo** [comment]
Method changelogs updated.
But I'm pretty sure you will rephrase them :D

**Comment by thomasplevy** [comment]
Nah they look good <3

**Pull request merged**